scripts/ingest_startgg.py

```diff
@@ -3,190 +3,216 @@
+                    "entrant_2_gamer_tags": entrant_id_to_gamer_tags[setObj["entrant2Id"]],
+                    "entrant_1_character_ids": setObj.get("entrant1CharacterIds", []),
+                    "entrant_2_character_ids": setObj.get("entrant2CharacterIds", []),
+                    "vod_url": setObj["vodUrl"]
+                })
+    return name, sets, date
+
+def set_tournament_vod_urls(slug: str, playlist_urls: list[tuple[str, str]], api_key: str):
+    print(f"Setting VOD URLs for {slug}")
     data = []
-    slugs = get_slugs_from_url(startgg_url)
-    if not slugs:
-        print("Could not get slugs")
-        return []
-    tournament_slug, event_slug = slugs
-    client = startgg_gql.get_client()
-    tournament_name, timezone = startgg_gql.get_tournament_name_and_timezone(client, tournament_slug)
-    event_id = startgg_gql.get_event_id(client, event_slug)
-    set_ids = startgg_gql.get_event_set_ids(client, event_id)
-    while len(set_ids):
-        set_id = set_ids[-1]
-        print(f"Processing set {set_id}")
-        try:
-            set_data = startgg_gql.get_set_data(client, set_id)
-            vod_data = get_vod_data(set_data, tournament_name, timezone)
-            if not vod_data:
-                # Try to get video from playlist
-                url = find_matching_youtube_video(set_data, playlist_urls)
-                if url:
-                    set_data["vodUrl"] = url
-                    vod_data = get_vod_data(set_data, tournament_name, timezone)
-                    try:
-                        startgg_gql.set_vod(client, set_id, url)
-                    except Exception as e:
-                        print(f"Failed to assign video: {e}")
-            if vod_data:
-                vod_data["id"] = len(data)
-                data.append(vod_data)
-            set_ids.pop()
-        except TransportServerError as err:
-            if err.code == 429:
-                print("Too many requests; taking a nap")
-                print("(You shouldn't see this if you're limiting reqs properly)")
-                client = startgg_gql.get_client()
-                time.sleep(10)
+    name, sets, date = get_tournament_name_sets_and_timezone(slug)
+
+    set_id_to_video_url = dict[str, str]()
+    current_playlist_urls = playlist_urls
+    unmatched_playlist_urls = []
+    while len(current_playlist_urls):
+        next_playlist_urls: list[tuple[str, str]] = []
+        for playlist_url in current_playlist_urls:
+            if playlist_url is None:
+                continue
+            video_title, video_url = playlist_url
+            video_title_safe = normalize(video_title)
+            exact_matching_sets = []
+            matching_sets = []
+            for setObj in sets:
+                all_gamer_tags_found = all([
+                    normalize(gamer_tag) in video_title_safe
+                    for gamer_tag in setObj["entrant_1_gamer_tags"] + setObj["entrant_2_gamer_tags"]
```

Comment:
I've been thinking this might should be more strict. I'm not sure of the best way to do this yet, since tags with spaces can be tough, but it might be best to split by word. A few pitfalls of this current method:

* Tags that are fully non-ASCII will always match, e.g. :sunglasses: vs A will match all sets that A is in
* Tags that are fully contained in another entrant's tag, e.g. "Player" will match all sets that "PlayerWithLongerTag" is in

This can be kicked down the road, though, since this is an existing problem. More-so this is just a reminder to me.

Reply:
I've added some more legal ASCII characters (that are youtube filename friendly), this helps Ble$$é not match Fable, for example. We can't do anything with spaces because youtube changes some filename chars into spaces automatically so we have to ignore spaces (EG. "s-f" becomes "s f").

I would like the emphasize that if you have round/phase name included, this is much less of a problem than you're probably thinking.